### DISCUSSION TITLE
#arc-enabled kubernetes - envoy lub app-controller gubi certyfikat/token ?

### DISCUSSION THREAD

**Discussion by sirkubax** [AzureCommunityPL/group#326] [Pomoc]
#arc-enabled kubernetes
Cześć, szukam gumowej kaczuszki, albo ktoś może już trafił na tego buga:
Po restarcie klastra, envoy w Arc-enabled kubernetes nie podnosi listenerów, bo mu się rozjeżdza certyfikat...
Nie mogę znaleźć który, bo wszystko co przejrzałem do tej pory wygląda poprawnie, działa poprawnie...

Jakieś pomysły przy podwieczorku? :D

Logi itp:
https://stackoverflow.com/questions/68544812/arc-enabled-kubernetes-envoy-looses-certificate-after-cluster-restart-and

**Comment by mgrabarz** [comment]
Podejdę do zagadnienia w oderwaniu od Arc, Kubetnetes, Envoy. 
Jaki jest powód błędu weryfikacji certyfikatu? Sprawdź co się wykłada jakims toolem które będzie się odpalać podczas startu podów. Typowo mogą to być:

- Brak synchronizacji czasu zanim klaster się ogarnie
- Brak dostępu do CA zanim sieć się ogarnie
- Błąd weryfikacji chain'a, zanim trusted CA się rozpropaguja
- Self signed zanim dojadą zaufane certy
- inne to usage, CA, SAN, cipher, ale tu nie sądzę :)

**Comment by flexray** [comment]
Tu jest wątek (może nie do końca ten sam, za to z tym samym błędem "encoding routines:asn1_d2i_read_bio:not enough data") o braku certa w odpowiednim miejscu (lub jak w tym wątku, montowany folder a nie konkretny plik) [asp-net-core-asn1-encoding-routinesasn1-d2i-read-bionot-enough-data-error-f](https://stackoverflow.com/questions/62863871/asp-net-core-asn1-encoding-routinesasn1-d2i-read-bionot-enough-data-error-f) - może jakoś pomoże, może nie jest dostępny lub ma złe uprawnienia po restarcie.
Pytanie skąd cert jest pobierany, jak z hosta to może potrzebny daemon-set który go po restarcie wkopiuje na node'a?

**Reply by sirkubax** [reply]
dzięki!
No własnie nie mogę namierzyć gdzie on siedzi, czy to env, czy jakiś sekret... zdaje mi się, ze jest to powiązane z custom-loaction, i działa po pierwsyzm deploy.. może to jakiś env.. ale jak mówisz, moze daemonset. Nie mogę namierzyć skąd to przychodzi, i gdzie powinno leżeć. Wszystkie miejsca które sprawdziłem wyglądają OK, certy które widzę są przed i po restarce... env, coś co znika, tak to możliwe scenariusze...
Pytałem już supportu MS, ludzi na forach... ten app-controller to nieco BlackBox, nie ma falgi do właczenia debug log :/ 
albo zgadne to przypadkiem, albo jedyna nadzieja że ticket trafi do architekta czy developera... 

Jose z MS, który pomaga obecnie, wydaje się rozumieć czym jest kubernetes, więc jest szansa! 
(w kontraście do tego, że mój poprzedni support ticket (inny temat) trafił do gościa który nie rozumiał co to jest Linux, to jest przepaść poznawcza).
